### PR TITLE
HOT-96682 Adding queue timeout with exponential backoff

### PR DESCRIPTION
**Pull request by mboudreau**
Also logging throttling events from octokit.

### CODE REVIEW

**Review by bgvozdev — APPROVED** [review]

File: src/worker/main.ts [hunk 1]
```diff
@@ -30,7 +30,12 @@ function measureElapsedTime(job: Queue.Job, tags) {
 
 const queueOpts: QueueOptions = {
 	defaultJobOptions: {
-		attempts: 3,
+		attempts: 5,
+		timeout: 60000,
+		backoff: {
+			type: "exponential",
+			delay: 3 * 60 * 1000
+		},
```
Comment (thombergs) [inline src/worker/main.ts:38]: can you add a feature flag around this change? Better safe than sorry

**Review by thombergs — CHANGES_REQUESTED** [review]

File: src/worker/main.ts [hunk 2]
```diff
@@ -30,7 +30,12 @@ function measureElapsedTime(job: Queue.Job, tags) {
 
 const queueOpts: QueueOptions = {
 	defaultJobOptions: {
-		attempts: 3,
+		attempts: 5,
+		timeout: 60000,
```
Comment (KAbakumovAtl) [inline src/worker/main.ts:34]: Wow, that looks too easy! :) 

Let's test it by simulating a long wait in src/config/enhance-octokit.ts:18

File: src/worker/main.ts (hunk 1 above)
Comment (KAbakumovAtl) [reply src/worker/main.ts:38]: I don't think it is feature flaggable, it can be only set on startup

**Review by KAbakumovAtl — APPROVED** [review]
Looks great, let's test it by simulating the long wait in the place where we have problems

**Review by thombergs — APPROVED** [review]

**Pull request merged**